piicatcher/generators.py

```diff
@@ -88,15 +92,22 @@ def _row_generator(
     column_list: List[CatColumn],
     sample_size=SMALL_TABLE_MAX,
 ):
-    engine = create_engine(source.conn_string)
+    if source.source_type == "bigquery":
+        engine = create_engine(source.conn_string, credentials_path=source.key_path)
+    else:
+        engine = create_engine(source.conn_string)
     with engine.connect() as conn:
-        dbinfo = get_dbinfo(source.source_type)
+        if source.source_type == "bigquery":
+            dbinfo = get_dbinfo(source.source_type, schema, table, source.project_id)
+        else:
+            dbinfo = get_dbinfo(source.source_type, schema, table)
```

Comment:
Instead of doing the special-case handling for BigQuery here, how about you just pass the catalog source object to `get_dbinfo` and let that function handle this?

```python
    def get_dbinfo(source: CatSource, schema: CatSchema, table: CatTable) -> Type[DbInfo]:
        source_type = source.source_type
        if source_type == "sqlite":
            return Sqlite(schema, table)
        ...
        elif source_type == "bigquery":
            return BigQuery(schema, table, source.project_id)
        raise AttributeError

    dbinfo = get_dbinfo(source, schema, table)
````

Reply:
Oh, also I think the return type of the `get_dbinfo` function needs to change from `Type[DbInfo]` to just `DbInfo`?